### PR TITLE
Add GAS/AT&T syntax support in asm.c assembler

### PR DESCRIPTION
**Pull request by Ruturaj4**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Added GAS/AT&T syntax support for the default asm.c assembler.

As required in https://github.com/rizinorg/rizin/issues/3256

...

**Test plan**

To reproduce:

```
./build/binrz/rz-asm/rz-asm -a x86 -s att "movl %eax, %ebx"
$89c3
```

...

**Closing issues**

...

### CODE REVIEW

**Review by wargio — COMMENTED** [review]
please add a test under `test/db/tools/rz_asm`

**Comment by Ruturaj4** [comment]
> please add a test under `test/db/tools/rz_asm`

ok done.

**Review by XVilka — CHANGES_REQUESTED** [review]
Code looks fine, please run `sys/clang-format.py` and commit (requires installing `clang-format` before-hand).

**Comment by Ruturaj4** [comment]
> Code looks fine, please run `sys/clang-format.py` and commit (requires installing `clang-format` before-hand).

done

File: librz/asm/asm.c [hunk 1]
```diff
@@ -725,12 +725,17 @@ RZ_API int rz_asm_assemble(RzAsm *a, RzAsmOp *op, const char *buf) {
...
+			// Check if the syntax is GAS/AT&T.
+			if (a->syntax == RZ_ASM_SYNTAX_ATT) {
+				ase = findAssembler(a, ".as");
+			} else {
+				/* find callback if no assembler support in current plugin */
+				ase = findAssembler(a, ".ks");
 				if (!ase) {
-					ase = findAssembler(a, NULL);
+					ase = findAssembler(a, ".nz");
+					if (!ase) {
```
Comment (DMaroo) [inline librz/asm/asm.c]: Bring this `if` condition one level outside, so as to avoid deep nesting. Something like

```c
if (!ase) {
	ase = ...;
}
if (!ase) {
	ase = ...;
}
```

**Review by DMaroo — CHANGES_REQUESTED** [review]

**Comment by DMaroo** [comment]
Use `clang-format-15` for the formatting. Refer https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#c.

> Use git-clang-format 15 to format your code. If clang-format-15 is not available on your Debian-based distribution, you can install it from https://apt.llvm.org/. You should invoke it as below (after making sure that your local copy of dev is up-to-date and your branch is up-to-date with dev):
> 
> ```git-clang-format-15 --extensions c,cpp,h,hpp,inc --style file dev```

**Comment by Ruturaj4** [comment]
> Use `clang-format-15` for the formatting. Refer https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#c.
> 
> > Use git-clang-format 15 to format your code. If clang-format-15 is not available on your Debian-based distribution, you can install it from https://apt.llvm.org/. You should invoke it as below (after making sure that your local copy of dev is up-to-date and your branch is up-to-date with dev):
> > `git-clang-format-15 --extensions c,cpp,h,hpp,inc --style file dev`

do you really want clang-format 15? I have latest (version 17), and I thought that should be fine

**Comment by wargio** [comment]
what OS are you using to have clang 17?

**Comment by Ruturaj4** [comment]
> 

I am using fedora, ohh, but i built it from the source

**Comment by wargio** [comment]
please use version 15

**Comment by Ruturaj4** [comment]
> please use version 15

ok sure thing. Once I will get some time, I will get that built.

**Comment by Ruturaj4** [comment]
> please use version 15

I just built and gave it a go, lets see if that works. thanks.

**Review by XVilka — APPROVED** [review]

File: librz/analysis/xrefs.c [hunk 2]
```diff
@@ -21,7 +21,8 @@ refs 10->20 C 16->10 J 20->10 C
 // XXX: is it possible to have multiple type for the same (from, to) pair?
 //      if it is, things need to be adjusted
 
-static RzAnalysisXRef *rz_analysis_xref_new(ut64 from, ut64 to, ut64 type) {
+static RzAnalysisXRef *
```
Comment (XVilka) [inline librz/analysis/xrefs.c]: Please remove this unrelated change from the PR

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/analysis/xrefs.c (hunk 2 above)
Comment (Ruturaj4) [reply librz/analysis/xrefs.c]: sure. Done.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Review by DMaroo — APPROVED** [review]

**Comment by XVilka** [comment]
It broke MacOS ARM64 and FreeBSD:
```
[XX] db/tools/rz_asm rz-asm -a x86 -s att "movl %eax, %ebx"
RZ_NOPLUGINS=1 /private/var/folders/bc/yvvw2g2s4jlg9r8x7tf61xv800009d/T/woodpecker-local-693079402/rizinorg/rizin/prefix/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc '!rz-asm -a x86 -s att "movl %eax, %ebx"' =
-- stdout
--- expected
+++ actual
@@ -1,1 +1,1 @@
-89c3
+00

-- stderr
ERROR: Please set 'RZ_X86_AS' env to define a 'x86' assembler program
ERROR: Please set 'RZ_X86_AS' env to define a 'x86' assembler program
ERROR: Please set 'RZ_X86_AS' env to define a 'x86' assembler program
```
